packages/dev/core/src/Meshes/Compression/dracoCodec.ts

```diff
@@ -0,0 +1,205 @@
+import type { Nullable } from "core/types";
+import { Tools } from "../../Misc/tools";
+import { AutoReleaseWorkerPool } from "../../Misc/workerPool";
+import type { IDisposable } from "../../scene";
+import { initializeWebWorker } from "./dracoCompressionWorker";
+
+/**
+ * Configuration for using a Draco codec.
+ */
+export interface IDracoCodecConfiguration {
+    /**
+     * The url to the WebAssembly module.
+     */
+    wasmUrl?: string;
+
+    /**
+     * The url to the WebAssembly binary.
+     */
+    wasmBinaryUrl?: string;
+
+    /**
+     * The url to the fallback JavaScript module.
+     */
+    fallbackUrl?: string;
+
+    /**
+     * The number of workers for async operations. Specify `0` to disable web workers and run synchronously in the current context.
+     */
+    numWorkers?: number;
+
+    /**
+     * Optional worker pool to use for async encoding/decoding.
+     * If provided, numWorkers will be ignored and the worker pool will be used instead.
+     * If provided the draco script will not be loaded from the DracoConfiguration.
+     */
+    workerPool?: AutoReleaseWorkerPool;
+
+    /**
+     * Optional ArrayBuffer of the WebAssembly binary.
+     * If provided it will be used instead of loading the binary from wasmBinaryUrl.
+     */
+    wasmBinary?: ArrayBuffer;
+
+    /**
+     * The codec module if already available.
+     */
+    jsModule?: any /* DracoDecoderModule | DracoEncoderModule */;
+}
+
+/**
+ * @internal
+ */
+export function GetDefaultNumWorkers(): number {
+    if (typeof navigator !== "object" || !navigator.hardwareConcurrency) {
+        return 1;
+    }
+
+    // Use 50% of the available logical processors but capped at 4.
+    return Math.min(Math.floor(navigator.hardwareConcurrency * 0.5), 4);
+}
+
+/**
+ * Base class for a Draco codec.
+ * @internal
+ */
+export abstract class DracoCodec<M> implements IDisposable {
+    protected _workerPoolPromise?: Promise<AutoReleaseWorkerPool>;
+    protected _modulePromise?: Promise<{ module: M }>;
+
+    /**
+     * The configuration for the Draco codec.
+     * Subclasses should override this value with a valid configuration.
+     */
+    public static Config: IDracoCodecConfiguration;
+
+    /**
+     * Returns true if the codec's `Configuration` is available.
+     */
+    public static get Available(): boolean {
+        return !!((this.Config.wasmUrl && this.Config.wasmBinaryUrl && typeof WebAssembly === "object") || this.Config.fallbackUrl);
+    }
+
+    /**
+     * The default draco compression object
+     * Subclasses should override this value with a valid instance.

```

Comment:
This statement isn't completely accurate. Subclasses need to create a public getter property.